spec/formatting.md

```diff
@@ -395,67 +411,73 @@ The resolution of _markup_ MUST always succeed.
 
 ### Fallback Resolution
 
-A **_<dfn>fallback value</dfn>_** is the _resolved value_ for an _expression_ that fails to resolve.
+A **_<dfn>fallback value</dfn>_** is the _resolved value_ for
+an _expression_ or _variable_ when that _expression_ or _variable_ fails to resolve.
+It contains a string representation that is used for its formatting,
+and no option values.
+
+The _resolved value_ of _text_, _literal_, and _markup_ MUST NOT be a _fallback value_.
+
+A _variable_ fails to resolve when no value is identified for its _name_.
+The string representation of its _fallback value_ is
+U+0024 DOLLAR SIGN `$` followed by the _name_ of the _variable_.
 
 An _expression_ fails to resolve when:
 
-- A _variable_ used as an _operand_ (with or without a _function_) fails to resolve.
-  * Note that this does not include a _variable_ used as an _option_ value.
-- A _function_ fails to resolve.
+- A _variable_ used as its _operand_ resolves to a _fallback value_.
+  Note that an _expression_ does not necessarily fail to resolve
+  if an _option_ resolves with a _fallback value_.
+- No _function handler_ is found for a _function_ _identifier_.
+- Calling a _function handler_ fails or does not return a valid value.
```

Comment:
Nitpicky editing. 

```suggestion
- The _function handler_ fails, returns an error, or returns an invalid value.
```

Reply:
Would an error not count as an invalid value?

Reply:
Maybe? It's unclear. It could be "validly an error" 😃 

This comment isn't that important. I'm mostly trying to harmonize phrasing here.

Reply:
I would much rather not re-introduce the phrase "returns an error", as that's rather opinionated about the implementation's processing model. We also should not use the more generic "emits an error", as it's possible for a function handler to emit an Unsupported Operation error and still return a valid value.

Reply:
Fair enough.